src/result_set.rs

```diff
@@ -295,12 +346,17 @@ fn parse_statistics(value: Value) -> RedisGraphResult<Statistics> {
 
 /// A scalar value returned by RedisGraph.
 #[derive(Debug, Clone, PartialEq)]
+#[non_exhaustive]
 pub enum Scalar {
```

Comment:
The term "scalar" has always kind of sucked (how is a string a scalar?), but with the addition of more complex data structures it's just misleading. Perhaps we should generalize this to something like "Type"?

Reply:
I agree in principle, however it still seems like `Scalar` is standard in most official implementations. It seems like a backwards-compatibility issue overall.

* Python: [ResultSetScalarTypes](https://github.com/RedisGraph/redisgraph-py/blob/94e263196055a7aa2dd85d41e7edc9f4561f2813/redisgraph/query_result.py#L15-L25)

* Java: [ResultSetScalarTypes](https://github.com/RedisGraph/JRedisGraph/blob/a44d01a74a340255f3fe735f352348e870f98d53/src/main/java/com/redislabs/redisgraph/impl/resultset/ResultSetScalarTypes.java)

* Go: [ResultSetScalarTypes](https://github.com/RedisGraph/redisgraph-go/blob/9890c26fbdd44f7c7c77641a22afb3774146febf/query_result.go#L34-L47)

In JavaScript and Ruby, the enum `ValueTypes` is used over `ScalarTypes`. However, all functions that return `ValueTypes` are still named `parse_scalar`, for example.

My main concern is the clash with the `redis` crate's `Value` enum which this library uses extensively.

Reply:
Welp, if it's common terminology, we should probably just leave it as is.

> Another thing to mention is how this library uses Relation instead of Edge. Would you like me to rename everything to Edge, since it seems like the standard terminology?

Yes, that's a good idea.